src/main/java/de/themorpheus/edu/taskservice/controller/solution/FreestyleSolutionController.java

```diff
@@ -52,10 +55,15 @@
 				optionalFreestyleSolution.get(),
 				dto.getSolution(),
 				UUID.randomUUID()
-			)
+				)
 		);
 
-		//TODO: PubSubService publish new Freestyle for teacher
+		this.messagePublisher.publish(new UserSentFreestyleSolutionDTO(
+				solutionResult.getResult().getTaskId().getAuthorId(),
+				UUID.randomUUID(),
```

Comment:
Don't u have the user id?

Reply:
Not yet, would be given by a param of the method or in the dto